### PR TITLE
added a basic service endpoint

### PR DESCRIPTION
**Pull request by sebastian**
@AndreiBozantan, I created the pull-request for you since the process wasn't clear when you pushed the code. But this pull-request still belongs to you.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            "serverReadyAction": {
+                "action": "openExternally",
+                "pattern": "^\\s*Now listening on:\\s+(https?://\\S+)"
+            },
+            "env": {
+                "ASPNETCORE_ENVIRONMENT": "Development"
+            },
+            "sourceFileMap": {
+                "/Views": "${workspaceFolder}/Views"
+            }
```
Comment (sebastian) [inline .vscode/launch.json]: There are no "views" in the project as far as I can tell? i.e. this is dead code?

File: Controllers/ExploreController.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        [HttpPost]
+        [Route("explore")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public IActionResult Explore(Models.ExploreParams data)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
```
Comment (sebastian) [inline Controllers/ExploreController.cs]: The errors produced are not very descriptive. Or rather they are descriptive but the problem is that the field names that are being complained about as missing do not match the field names as they would be provided in the input JSON.

Here is an example from the tests you provided in slack:

```
- wrong data in request JSON (missing api_key)
> curl -k -X POST -H "Content-Type: application/json" https://localhost:5001/explore  -d '{"table_name":"t1", "column_name":"c2"}'
{"type":"https://tools.ietf.org/html/rfc7231#section-6.5.1","title":"One or more validation errors occurred.","status":400,"traceId":"|ecdc1210-4b8aaa2689f85747.","errors":{"ApiKey":["The ApiKey field is required."]}}
```

Notice that the request should have contained `api_key` but your error complains about `ApiKey`.

File: Program.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
```
Comment (sebastian) [inline Program.cs]: Most of this imports aren't actually needed/required. VSCode tells me we only actually need:

```
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Hosting;
```

File: Properties/launchSettings.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  "profiles": {
+    "IIS Express": {
+      "commandName": "IISExpress",
+      "launchBrowser": true,
+      "launchUrl": "weatherforecast",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
```
Comment (sebastian) [inline Properties/launchSettings.json]: The default config in ASP.NET Core uses [kestrel](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-3.1) which is the high performance cross platform web server shipping with dotnet core. We are not going to use IIS. Hence this seems dead code too?

File: Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "Explorer": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "launchUrl": "weatherforecast",
```
Comment (sebastian) [inline Properties/launchSettings.json]: What is "weatherforecast"? It seems a remnant from copying and pasting?

File: Startup.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.HttpsPolicy;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
```
Comment (sebastian) [inline Startup.cs]: Again quite a number of unnecessary imports. The only ones used are: 

```cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
```

File: Startup.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
```
Comment (sebastian) [inline Startup.cs]: This comment isn't very useful. I would remove it.

File: Startup.cs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllers();
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
```
Comment (sebastian) [inline Startup.cs]: Same here

**Comment by sebastian** [comment]
Also, as a more general comment: we are going to need multiple dotnet core projects. Probably some class libraries, some console applications for tests etc. Isn't it common to put all the individual apps under a `src/Component` directory, and then have a top level solution file?

File: .vscode/launch.json (hunk 1 above)
Comment (AndreiBozantan) [reply .vscode/launch.json]: I removed that.

File: Controllers/ExploreController.cs (hunk 2 above)
Comment (AndreiBozantan) [reply Controllers/ExploreController.cs]: I did some improvements on that, but it is still not ideal. I will try to search some other solutions to fix this completely.

File: Program.cs (hunk 3 above)
Comment (AndreiBozantan) [reply Program.cs]: I removed the unused imports.

File: Properties/launchSettings.json (hunk 4 above)
Comment (AndreiBozantan) [reply Properties/launchSettings.json]: I cleaned up the launchSettings.json file.

File: Properties/launchSettings.json (hunk 5 above)
Comment (AndreiBozantan) [reply Properties/launchSettings.json]: I cleaned up the launchSettings.json file.

File: Startup.cs (hunk 6 above)
Comment (AndreiBozantan) [reply Startup.cs]: I removed the unused imports.

File: Startup.cs (hunk 7 above)
Comment (AndreiBozantan) [reply Startup.cs]: Cleaned up.

File: Startup.cs (hunk 8 above)
Comment (AndreiBozantan) [reply Startup.cs]: Cleaned up.

File: Controllers/ExploreController.cs (hunk 2 above)
Comment (sebastian) [reply Controllers/ExploreController.cs]: Before we go too far down this route: If there is no simple obvious solution to this problem, then I am totally happy with changing the parameter names in the API instead! I.e. have the payload be: 

```json
{
  "ApiKey": "...",
  "TableName": "...",
  "ColumnName": "..."
}
```

instead.

That way the problem kind of solved itself?

**Pull request closed without merging**